web/server/codechecker_server/server.py

```diff
@@ -319,31 +319,37 @@ def do_POST(self):
         """
         Handles POST queries, which are usually Thrift messages.
         """
+        protocol_factory = TJSONProtocol.TJSONProtocolFactory()
+        input_protocol_factory = protocol_factory
+        output_protocol_factory = protocol_factory
+
+        # Get Thrift API function name to print to the log output.
+        itrans = TTransport.TFileObjectTransport(self.rfile)
+        itrans = TTransport.TBufferedTransport(itrans,
+                                               int(self.headers[
+                                                   'Content-Length']))
+        iprot = input_protocol_factory.getProtocol(itrans)
+        fname, _, _ = iprot.readMessageBegin()
+
         client_host, client_port, is_ipv6 = \
             RequestHandler._get_client_host_port(self.client_address)
         self.auth_session = self.__check_session_cookie()
-        LOG.info("%s:%s -- [%s] POST %s",
+        LOG.info("%s:%s -- [%s] POST %s@%s",
                  client_host if not is_ipv6 else '[' + client_host + ']',
                  client_port,
                  self.auth_session.user if self.auth_session else "Anonymous",
-                 self.path)
+                 self.path, fname)
 
         # Create new thrift handler.
         checker_md_docs = self.server.checker_md_docs
         checker_md_docs_map = self.server.checker_md_docs_map
         version = self.server.version
 
-        protocol_factory = TJSONProtocol.TJSONProtocolFactory()
-        input_protocol_factory = protocol_factory
-        output_protocol_factory = protocol_factory
+        cstringio_buf = itrans.cstringio_buf.getvalue()
+        itrans = TTransport.TMemoryBuffer(cstringio_buf)
+        iprot = input_protocol_factory.getProtocol(itrans)
```

Comment:
This is needed because previously we started reading when `iprot.readMessageBegin()` was called, right? I hope this "low-level access" won't backfire magically.